pandas/tests/io/parser/test_network.py

```diff
@@ -15,42 +15,36 @@
 from pandas.io.parsers import read_csv, read_table
 
 
-class TestCompressedUrl(object):
+@pytest.fixture(scope='module')
+def salaries_table():
+    path = os.path.join(tm.get_data_path(), 'salaries.csv')
+    return read_table(path)
 
-    compression_to_extension = {
-        'gzip': '.gz',
-        'bz2': '.bz2',
-        'zip': '.zip',
-        'xz': '.xz',
-    }
 
-    def setup(self):
-        path = os.path.join(tm.get_data_path(), 'salaries.csv')
-        self.local_table = read_table(path)
-        self.base_url = ('https://github.com/pandas-dev/pandas/raw/master/'
-                         'pandas/io/tests/parser/data/salaries.csv')
+@pytest.mark.parametrize(
+    "compression,extension", [('gzip', '.gz'), ('bz2', '.bz2'),
+                              ('zip', '.zip'), ('xz', '.xz')])
+def test_compressed_urls(salaries_table, compression, extension):
```

Comment:
so you just use ``decorator.decorator`` instead of ``functools.wraps``?

if its only for testing purposes, then we could easily do that (though even easier would be to fix our ``tm.network`` ....)